### PR TITLE
coordinator: support for stop conflation at block timestamp

### PR DESCRIPTION
**Pull request by fluentcrafter**
This PR implements issue(s) #

### Checklist

* [ ] I wrote new tests for my new core changes.
* [ ] I have successfully ran tests, style checker and build against my new changes locally.
* [ ] I have informed the team of any breaking changes if there are any.

---

> [!NOTE]
> Adds support to stop conflation at a target L2 block timestamp and migrates block monitoring to the new EthApi client, with TOML parsing for Instants and updated tests.
> 
> - **Conflation config and TOML**:
>   - Add `conflation.forceStopConflationAtBlockTimestampInclusive: Instant?` in `ConflationConfig` and `ConflationToml`, and propagate to runtime.
>   - Use `kotlinx.datetime.Instant` across configs; remove conversions.
> - **Config loading**:
>   - Add `TomlKotlinInstantDecoder` to parse ISO-8601 strings or epoch seconds and register it in `ConfigLoaderHelper`.
> - **Block monitoring/runtime**:
>   - `BlockCreationMonitor` now uses `EthApiBlockClient` instead of `ExtendedWeb3J` and fetches blocks via `ethGetBlockByNumberFullTxs`.
>   - Support stopping when reaching `lastL2BlockTimestampToProcessInclusive` or `lastL2BlockNumberToProcessInclusive`; stop service on reorg or fetch-limit overflow.
>   - `ConflationApp` creates `EthApiClient` and passes new timestamp stop config to `BlockCreationMonitor`.
> - **Tests**:
>   - Extend parsing tests to cover new timestamp fields and epoch-seconds parsing.
>   - Update `BlockCreationMonitorTest` to use `EthApiClient`, generate timestamps, and add test to stop at timestamp boundary.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 57967d262275fce453dcefa9770fd8233c194f27. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: coordinator/app/src/main/kotlin/net/consensys/zkevm/coordinator/blockcreation/BlockCreationMonitor.kt [hunk 1]
```diff
@@ -120,12 +123,25 @@ class BlockCreationMonitor(
             config.lastL2BlockNumberToProcessInclusive,
             nexBlockNumberToFetch,
           )
+          this.stop()
```
Comment (gauravahuja) [inline coordinator/app/src/main/kotlin/net/consensys/zkevm/coordinator/blockcreation/BlockCreationMonitor.kt:126]: why call this.stop here vs super.stop on line 143 and 164?

File: coordinator/app/src/main/kotlin/net/consensys/zkevm/coordinator/blockcreation/BlockCreationMonitor.kt [hunk 2]
```diff
@@ -109,6 +111,7 @@ class BlockCreationMonitor(
             _nexBlockNumberToFetch.get() - lastProvenBlockNumber,
             config.blocksFetchLimit,
           )
+          this.stop()
```
Comment (gauravahuja) [inline coordinator/app/src/main/kotlin/net/consensys/zkevm/coordinator/blockcreation/BlockCreationMonitor.kt:114]: stop is an async method, so you can return the safefuture it returns instead of Safefuture.COMPLETE

File: coordinator/app/src/main/kotlin/net/consensys/zkevm/coordinator/blockcreation/BlockCreationMonitor.kt (hunk 1 above)
Comment (fluentcrafter) [reply coordinator/app/src/main/kotlin/net/consensys/zkevm/coordinator/blockcreation/BlockCreationMonitor.kt:126]: we must favour this.stop over super.stop, because if stop is overrided it will not work properly

File: coordinator/app/src/main/kotlin/net/consensys/zkevm/coordinator/blockcreation/BlockCreationMonitor.kt (hunk 2 above)
Comment (fluentcrafter) [reply coordinator/app/src/main/kotlin/net/consensys/zkevm/coordinator/blockcreation/BlockCreationMonitor.kt:114]: That would fail the InterruptedException. so it should be this way
